### ISSUE TITLE
Rust symbol demangling

### ISSUE THREAD

**Issue by adammitha** [bpftrace/bpftrace#3532]
### Is your feature request related to a problem? Please describe.
`bpftrace` currently doesn't support demangling Rust symbols, so when listing `uprobes` on Rust programs you get the mangled symbol name. This is a suboptimal user experience.

Unfortunately, Rust's name mangling scheme is different from C++, so the `:cpp` prefix doesn't work on Rust binaries.

For example:
```
❯ bpftrace --version
bpftrace v0.20.4
adam @ adams-nixos-desktop 🌐in nixos-configuration on  main
❯ sudo bpftrace -l 'u:/nix/store/71jzy8l7bqkw64fn4gnldrr248vqi2ly-ripgrep-14.1.0/bin/rg:cpp:*' | head -n 1
uprobe:/nix/store/71jzy8l7bqkw64fn4gnldrr248vqi2ly-ripgrep-14.1.0/bin/rg:cpp:"_$LT$aho_corasick..packed..teddy..builder..x86_64..SlimAVX2$LT$_$GT$$u20$as$u20$core..fmt..Debug$GT$::fmt::h5c8301c85bfe80c1"
```

### Describe the solution you'd like
Introduce a `:rust` prefix to the `bpftrace` probe syntax that allows users to request automatic symbol demangling for Rust symbols in the same way that they request automatic symbol demangling.

LLVM has a rust demangler that we _should_ be able to use: https://github.com/llvm/llvm-project/blob/main/llvm/lib/Demangle/RustDemangle.cpp.

I'm happy to do the work, but since this would be my first contribution I wanted to discuss the idea before implementing it.

### Describe alternative solutions or features you've considered
Use the mangled symbol names – not a great user experience.

**Comment by amiremohamadi** [comment]
not quite sure, but i think we can provide a general `demangle` prefix for both ([Demangle.cpp](https://github.com/llvm/llvm-project/blob/11dad2fa5138a50d60a5a34a2c7e074b976820e2/llvm/lib/Demangle/Demangle.cpp#L46))

**Comment by amiremohamadi** [comment]
as far as I know, LLVM has supported rust demangling since release/13.x. therefore, we can rely on LLVM to automatically use the appropriate rust demangler (or another one if necessary) when we apply the demangle prefix.

**Comment by adammitha** [comment]
Oh good call - I'll give that a shot.

**Comment by batu3H** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by amiremohamadi** [comment]
quick question: AOT runtime won’t support rust demangling since it relies on the [stdlib demangler, not LLVM](https://github.com/bpftrace/bpftrace/commit/901e13526c842e04cecb15d312317fc7f392b863). is it fine to skip rust demangling in AOT?

cc: @jordalgo @danobi

**Comment by jordalgo** [comment]
I think [blazesym](https://github.com/libbpf/blazesym), which we now use, can demangle Rust symbols. Keep me honest here @d-e-s-o

**Comment by d-e-s-o** [comment]
> I think [blazesym](https://github.com/libbpf/blazesym), which we now use, can demangle Rust symbols. Keep me honest here [@d-e-s-o](https://github.com/d-e-s-o)

Yes, you are correct. It should be enabled by default when using `blazesym` for symbolization.

Edit: By "by-default" I mean that there is no opt-out for this support at the `blazesym` level. For `bpftrace` you'd need to make sure that `cpp_demangle` is active, as that's the knob that [we honor](https://github.com/bpftrace/bpftrace/blob/679ba221156a8337064f2b4426b25673bf7fda3d/src/usyms.cpp#L167).

**Comment by jordalgo** [comment]
It looks like `cpp_demangle` defaults to true in the bpftrace config. So perhaps Rust demangling already works when Blazesym is used.

**Comment by flxzt** [comment]
Rust now defaults to it's new mangling scheme "v0" (see: https://blog.rust-lang.org/2026/07/09/Rust-1.97.0/). There is a library available that helps with demangling: https://github.com/rust-lang/rustc-demangle
